### PR TITLE
OPHJOD-290: PathProgress component

### PR DESCRIPTION
**Pull request by juhaniko**
## Description

* Initial version for PathProgress. There are still design/functionality questions to be resolved before it can be finalized.

## Related JIRA ticket

https://jira.eduuni.fi/browse/OPHJOD-290
image

### CODE REVIEW

File: lib/components/PathProgress/PathProgress.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import { cx } from '../../cva';
+
+export interface PathProgressProps {
+  steps: number;
+  currentStep: number;
+}
+
+const getButtonClasses = (active: boolean, idx: number, isIcon = false) =>
+  cx('size-8 rounded-full appearance-none z-10 mb-6', {
+    'bg-accent !text-white': active,
```
Comment (ketsappi) [inline lib/components/PathProgress/PathProgress.tsx:11]: `!text-white`
Would be nice not to use `!important` in design system.

One way might be to have own class for the actual text/icon and separate the logic from the background.

Also the `!mb-0` would be nice to done without `!important`

Comment (juhaniko) [reply lib/components/PathProgress/PathProgress.tsx:11]: I was wondering why the classes don't get applied without important, and the culprit was `cx` combined with `twMerge`. They would omit `text-white` if `text-button-lg` was used. Also the order of the object properties in `cx` also affected the inclusion of css classes.

File: lib/components/PathProgress/PathProgress.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { cx } from '../../cva';
+
+export interface PathProgressProps {
```
Comment (ketsappi) [inline lib/components/PathProgress/PathProgress.tsx:4]: Maybe there should be `onClick` prop as these seems to be buttons. And it is propably wanted that user can click these path steps to jump to clicked one 🤔

Comment (juhaniko) [reply lib/components/PathProgress/PathProgress.tsx:4]: Yeah, I'm implementing right now

Comment (ketsappi) [reply lib/components/PathProgress/PathProgress.tsx:4]: Now it is not as a prop as I originally meant, but the current assumption is good and the best guess we have how this component is actually used.
Let's have it like this.

**Review by ketsappi — APPROVED** [review]

**Pull request merged**